pkg/controlplane/controller.go

```diff
@@ -236,141 +163,3 @@ func (c *Controller) RunKubernetesNamespaces(ch chan struct{}) {
 		}
 	}, c.SystemNamespacesInterval, ch)
 }
-
-// RunKubernetesService periodically updates the kubernetes service
-func (c *Controller) RunKubernetesService(ch chan struct{}) {
-	// wait until process is ready
-	wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
-		var code int
-		c.client.CoreV1().RESTClient().Get().AbsPath("/readyz").Do(context.TODO()).StatusCode(&code)
-		return code == http.StatusOK, nil
-	}, ch)
-
-	wait.NonSlidingUntil(func() {
-		// Service definition is not reconciled after first
-		// run, ports and type will be corrected only during
-		// start.
-		if err := c.UpdateKubernetesService(false); err != nil {
```

Comment:
actually Service definition seems is never reconciled since https://github.com/kubernetes/kubernetes/commit/2a9a9fa155e92d0bd7f4eb85d843cfa14b830cca

Reply:
Have you followed up on this?  That would be... "bad"?